test/examples/globe.js

```diff
@@ -33,4 +34,61 @@ describe('globe', () => {
         page.close();
         await page.close();
     });
+    it('should not add layers beyond the capabilities', async function _() {
+        const page = await browser.newPage();
+
+        await page.setViewport({ width: 400, height: 300 });
+        await page.goto(`http://localhost:${itownsPort}/examples/globe.html`);
+        await page.waitFor('#viewerDiv > canvas');
+
+        await exampleCanRenderTest(page, this.test.fullTitle());
+
+        const maxColorSamplerUnitsCount = await page
+            .evaluate(type => globeView.wgs84TileLayer.level0Nodes[0]
+                .material.textures[type].length, lCOLOR);
+        const colorSamplerUnitsCount = await page.evaluate(() =>
+                globeView.wgs84TileLayer.countColorLayersTextures([], globeView.getLayers(l => l.type === 'color')[0]));
+        const limit = maxColorSamplerUnitsCount - colorSamplerUnitsCount;
+
+        // add layers just below the capacity limit
+        const underLimit = await page.evaluate(maxLayersCount =>
+            itowns.Fetcher.json('./layers/JSONLayers/OrthosCRS.json').then((params) => {
+                const promises = [];
+                for (let i = 0; i < maxLayersCount; i++) {
+                    const layerParams = Object.assign({}, params);
+                    layerParams.id = `${layerParams.id}_${i}`;
+                    promises.push(globeView.addLayer(layerParams));
+                }
+                return Promise.all(promises).then(() => true).catch(() => false);
+            }), limit);
+
+        // add one layer just over the capacity limit
+        // verify if the error is handled
+        const errorOverLimit = await page.evaluate(() =>
+            itowns.Fetcher.json('./layers/JSONLayers/OrthosCRS.json').then((params) => {
+                const layerParams = Object.assign({}, params);
+                layerParams.id = 'max';
+                return globeView.addLayer(layerParams).then(() => false).catch(() => true);
+            }));
+
+        assert.ok(underLimit);
```

Comment:
The `assert` could be moved above (just after `underLimit` is assigned).

Reply:
why?